src/test/metrics_test.cpp

```diff
@@ -0,0 +1,414 @@
+    metric->RecordEntry(0.7128684686471212, 12.180350761995374);
+    metric->RecordEntry(0.7250488194091166, 11.849875908074102);
+    metric->RecordEntry(0.7368986953171907, 11.25893134410846);
+    metric->RecordEntry(0.7481576266612991, 11.29556518338785);
+    metric->RecordEntry(0.759453191844687, 12.100159818336204);
+    metric->RecordEntry(0.7715533516630232, 11.268292831597137);
+    metric->RecordEntry(0.7828216444946203, 10.876697084092664);
+    metric->RecordEntry(0.793698341578713, 11.41067769966513);
+    metric->RecordEntry(0.8051090192783781, 10.764130551471954);
+    metric->RecordEntry(0.81587314982985, 11.516565679587151);
+    metric->RecordEntry(0.8273897155094372, 12.258346383658813);
+    metric->RecordEntry(0.839648061893096, 12.042792906933066);
+    metric->RecordEntry(0.8516908548000292, 12.006355840344012);
+    metric->RecordEntry(0.8636972106403732, 12.436359135554602);
+    metric->RecordEntry(0.8761335697759278, 11.877641201985803);
+    metric->RecordEntry(0.8880112109779136, 10.567107149500245);
+    metric->RecordEntry(0.8985783181274138, 11.911189275994568);
+    metric->RecordEntry(0.9104895074034084, 10.889245781607388);
+    metric->RecordEntry(0.9213787531850158, 12.465256618853836);
+    metric->RecordEntry(0.9338440098038696, 11.223065556424801);
+    metric->RecordEntry(0.9450670753602944, 11.730574094963503);
+    metric->RecordEntry(0.9567976494552579, 11.325378236493892);
+    metric->RecordEntry(0.9681230276917518, 11.874416573715038);
+    metric->RecordEntry(0.9799974442654669, 11.868266253898241);
+    metric->RecordEntry(0.9918657105193651, 10.700628941411287);
+    metric->RecordEntry(1.0025663394607764, 11.199358872070935);
+    metric->RecordEntry(1.0137656983328474, 10.863209130580797);
+    metric->RecordEntry(1.024628907463428, 12.04804735335845);
+    metric->RecordEntry(1.0366769548167865, 12.060037474628244);
+    metric->RecordEntry(1.0487369922914147, 12.07516039943628);
+    metric->RecordEntry(1.060812152690851, 11.636401196731727);
+    metric->RecordEntry(1.0724485538875828, 11.325091973540063);
+    metric->RecordEntry(1.083773645861123, 12.367583932075869);
+    metric->RecordEntry(1.0961412297931987, 12.100536564029815);
+    metric->RecordEntry(1.1082417663572286, 10.709781403850688);
+    metric->RecordEntry(1.1189515477610792, 11.965315843361148);
+    metric->RecordEntry(1.1309168636044404, 11.3946157918795);
+    metric->RecordEntry(1.1423114793963198, 11.310652174786224);
+
+    const metrics::GaugeBasicStatistics basicStatistics = metric->GetBasicStatistics();
+    EXPECT_EQ(basicStatistics.min, 10.50685559634115);
+    EXPECT_EQ(basicStatistics.max, 12.465256618853836);
+    EXPECT_DOUBLE_EQ(basicStatistics.average, 11.53622131571106);
+    EXPECT_DOUBLE_EQ(basicStatistics.timeRatio, 1009.9015482018648);
+
+    const metrics::GaugeComplexStatistics complexStatistics = metric->ComputeComplexStatistics();
+    EXPECT_EQ(complexStatistics.median, 11.526757950183075);
```

Comment:
Seems like below you used the approximate EXPECT_DOUBLE_EQ, so why not here?

Reply:
Ah good catch. This one does actually lead to a computation when the number of entries is even. Fixed.